spring-integration-jms/src/main/java/org/springframework/integration/jms/dsl/Jms.java

```diff
@@ -0,0 +1,325 @@
+	}
+
+	/**
+	 * The factory to produce a {@link JmsOutboundGatewaySpec}.
+	 * @param connectionFactory the JMS ConnectionFactory to build on
+	 * @return the {@link JmsOutboundGatewaySpec} instance
+	 */
+	public static JmsOutboundGatewaySpec outboundGateway(ConnectionFactory connectionFactory) {
+		return new JmsOutboundGatewaySpec(connectionFactory);
+	}
+
+	/**
+	 * The factory to produce a {@link JmsOutboundGatewaySpec}.
+	 * @param listenerContainer the JMS {@link AbstractMessageListenerContainer} to build on
+	 * @param <S> the {@link JmsInboundGatewaySpec} inheritor type
+	 * @return the {@link JmsOutboundGatewaySpec} instance
+	 */
+	public static <S extends JmsInboundGatewaySpec<S>> JmsInboundGatewaySpec<S> inboundGateway(
+			AbstractMessageListenerContainer listenerContainer) {
+		return new JmsInboundGatewaySpec<S>(listenerContainer);
+	}
+
+	/**
+	 * The factory to produce a {@link JmsOutboundGatewaySpec}.
+	 * @param connectionFactory the JMS ConnectionFactory to build on
+	 * @return the {@link JmsOutboundGatewaySpec} instance
+	 */
+	public static JmsInboundGatewaySpec.JmsInboundGatewayListenerContainerSpec<JmsDefaultListenerContainerSpec, DefaultMessageListenerContainer>
+	inboundGateway(ConnectionFactory connectionFactory) {
+		return inboundGateway(connectionFactory, DefaultMessageListenerContainer.class);
+	}
+
+	/**
+	 * The factory to produce a {@link JmsOutboundGatewaySpec}.
+	 * @param connectionFactory the JMS ConnectionFactory to build on
+	 * @param containerClass    the {@link AbstractMessageListenerContainer} implementation class
+	 *                          to instantiate listener container
+	 * @param <S>               the {@link JmsListenerContainerSpec} inheritor type
+	 * @param <C>               the {@link AbstractMessageListenerContainer} inheritor type
+	 * @return the {@link JmsOutboundGatewaySpec} instance
+	 */
+	public static <S extends JmsListenerContainerSpec<S, C>, C extends AbstractMessageListenerContainer>
+	JmsInboundGatewaySpec.JmsInboundGatewayListenerContainerSpec<S, C> inboundGateway(ConnectionFactory connectionFactory,
+			Class<C> containerClass) {
+		try {
+			JmsListenerContainerSpec<S, C> spec =
+					new JmsListenerContainerSpec<S, C>(containerClass)
+							.connectionFactory(connectionFactory);
+			return new JmsInboundGatewaySpec.JmsInboundGatewayListenerContainerSpec<S, C>(spec);
+		}
+		catch (Exception e) {
+			throw new IllegalStateException(e);
+		}
+	}
+
+	/**
+	 * The factory to produce a {@link JmsMessageDrivenChannelAdapterSpec}.
+	 * @param listenerContainer the {@link AbstractMessageListenerContainer} to build on
+	 * @param <S>               the {@link JmsMessageDrivenChannelAdapterSpec} inheritor type
+	 * @return the {@link JmsMessageDrivenChannelAdapterSpec} instance
+	 * @deprecated - use {@link #messageDrivenChannelAdapter(AbstractMessageListenerContainer)}.
+	 */
+	@Deprecated
```

Comment:
We don't need these `deprecated` method since we are in the major version already.
Fixing on merge...